implementations/rust/ockam/ockam_node/src/context/send_message.rs

```diff
@@ -280,35 +280,44 @@ impl Context {
         Ok(())
     }
 
+    /// Forward a transport message to its next routing destination
+    /// This removes the message first onward address (considered as processed)
+    /// and appends the current Context address to the message return route
+    pub async fn forward(&self, local_msg: LocalMessage) -> Result<()> {
```

Comment:
I think we need a clear definitions around `forward` and `send`. Previously:
1. `send` is to create a new message and send from the current address
2. `forward` move an existing message along the route. Also, caller is responsible for moving all the Addresses inside onward and return route

In this PR I'm concerned it's no longer clear what's exactly the logic behind the naming

Reply:
Maybe, everything that takes a `LocalMessage` should be `forward` and all the addresses should be manipulated by the caller prior to calling `forward`. Thoughts?

Reply:
I'm also concerned that we change the behaviour of the previously existing function with the exact same name. It's easy to beak something this way

Reply:
I reverted back the naming. I think I am annoyed that we have 2 APIs for sending messages, one with `send` and one with `forward`, especially with a comment that says that `forward` is low-level, yet used in something as simple as a `Hop` worker. I'm going to leave it like it was for now. Hopefully we can come back to this API.